solc/CommandLineParser.cpp

```diff
@@ -977,9 +995,28 @@ void CommandLineParser::processArgs()
 	for (auto& option: conflictingWithStopAfter)
 		checkMutuallyExclusive({g_strStopAfter, option});
 
+	array<string, 12> const conflictingWithAsmJsonImport{
+		g_strOptimize,
+		CompilerOutputs::componentName(&CompilerOutputs::ir),
+		CompilerOutputs::componentName(&CompilerOutputs::irOptimized),
+		CompilerOutputs::componentName(&CompilerOutputs::ewasm),
+		CompilerOutputs::componentName(&CompilerOutputs::ewasmIR),
+		g_strGas,
+		CompilerOutputs::componentName(&CompilerOutputs::metadata),
+		CompilerOutputs::componentName(&CompilerOutputs::natspecDev),
+		CompilerOutputs::componentName(&CompilerOutputs::natspecUser),
+		CompilerOutputs::componentName(&CompilerOutputs::signatureHashes),
+		CompilerOutputs::componentName(&CompilerOutputs::storageLayout),
+		CompilerOutputs::componentName(&CompilerOutputs::astCompactJson),
+	};
```

Comment:
Since this option is one of the input modes, you don't need to list outputs here. `parseOutputSelection()` will take care of them.
```suggestion
	array<string, 2> const conflictingWithEVMAsmImport{
		g_strOptimize,
		g_strGas,
	};
```

Reply:
I added a more restrictive check here that is based on all supported command-line parameters in this mode. See `CommandLineParser::processArgs()` search for `supportedByEvmAsmJsonImport`.